app/client/components/helpCentre/knownIssues.tsx

```diff
@@ -0,0 +1,119 @@
+import { css } from "@emotion/core";
+import { breakpoints, palette, space } from "@guardian/src-foundations";
+import { textSans } from "@guardian/src-foundations/typography";
+import React, { useEffect, useState } from "react";
+import { ProductDetail } from "../../../shared/productResponse";
+import { allProductsDetailFetcher } from "../../../shared/productTypes";
+import { minWidth } from "../../styles/breakpoints";
+import { gridBase, gridItemPlacement } from "../../styles/grid";
+import { ErrorIcon } from "../svgs/errorIcon";
+
+interface IdentityDetailsWithStatus extends IdentityDetails {
+  signInStatus?: string;
+}
+
+interface Issue {
+  date: string;
+  message: string;
+  affectedProducts?: string[];
+}
+
+export const KnownIssues = () => {
+  const [issuesData, setIssuesData] = useState<Issue[]>([]);
+  useEffect(() => {
+    (async () => {
+      const knownIssuesResponse = await fetch("/api/known-issues/");
+      const unfilteredKnownIssues: Issue[] = knownIssuesResponse.ok
+        ? await knownIssuesResponse.json()
+        : [];
+      const unfilteredDateSortedIssues = unfilteredKnownIssues.sort(
+        (a, b) => Date.parse(a.date) - Date.parse(b.date)
+      );
+      const responseContainsProductIssues = unfilteredKnownIssues.some(
```

Comment:
`unfilteredKnownIssues` will always be an extremely small array. For that reason you could remove `responseContainsProductIssues` and have a cleaner code. I believe the benefits of cleaner code outweigh the performance benefits of having this var.

Reply:
`responseContainsProductIssues` here is used to check if we have any product specific issues as opposed to the generic global issues.

If `responseContainsProductIssues` is true then an additional api call to members-data-api is necessary in order to match the product specific issues up with the products that the user has.

`responseContainsProductIssues` is therefore a gate between making an unnecessary api call if there are only global issues.